packages/foam-vscode/src/features/folder-rename-dialog.ts

```diff
@@ -0,0 +1,153 @@
+      });
+    } else {
+      items.push({
+        label: '$(check) Proceed',
+        detail: dryRunResult ? `No link updates needed - proceed with the rename (${this.pluralize(dryRunResult.filesProcessedCount, 'file')} checked).` : 'Proceed with the rename.'
+      });
+    }
+
+    items.push({
+      label: '$(x) Skip Link Updates',
+      detail: dryRunResult && dryRunResult.linksUpdatedCount > 0
+        ? `Rename the folder but leave ${this.pluralize(dryRunResult.linksUpdatedCount, 'link')} unchanged.`
+        : 'Rename the folder but don\'t update any links.'
+    });
+
+    items.push({
+      label: '$(discard) Abort Rename',
+      detail: `Revert the folder name back to '${oldPath}' and cancel the entire operation.`
+    });
+
+    // Add settings option only if currently in "confirm" mode
+    if (currentMode === 'confirm') {
+      items.push({
+        label: '$(gear) Settings',
+        detail: 'Change folder rename behavior in Foam settings.'
+      });
+    }
+
+    const selected = await new Promise<vscode.QuickPickItem | undefined>((resolve) => {
+      const quickPick = vscode.window.createQuickPick();
+
+      // Enhanced title with dry run results, simplified placeholder
+      if (dryRunResult) {
+        if (dryRunResult.linksUpdatedCount > 0) {
+          quickPick.title = `Foam: Folder Rename - ${this.pluralize(dryRunResult.linksUpdatedCount, 'link')} to update`;
+        } else {
+          quickPick.title = `Foam: Folder Rename - No links to update`;
+        }
+      } else {
+        quickPick.title = 'Foam: Folder Rename';
+      }
+
+      quickPick.placeholder = 'Choose how to handle this folder rename...';
+      quickPick.ignoreFocusOut = true;
+      quickPick.matchOnDetail = true;
+      quickPick.items = items;
+
+      quickPick.onDidAccept(() => {
+        const selection = quickPick.selectedItems[0];
+        quickPick.hide();
+        resolve(selection);
+      });
+
+      quickPick.onDidHide(() => {
+        quickPick.dispose();
+        resolve(undefined);
+      });
+
+      quickPick.show();
+
+      // Force focus to the QuickPick after a short delay
+      setTimeout(async () => {
+        try {
+          // Try to focus the QuickPick directly
+          await vscode.commands.executeCommand('workbench.action.focusQuickOpen');
+        } catch {
+          Logger.debug('Could not force focus to QuickPick dialog');
+        }
+      }, 50);
+    });
+
+    if (!selected) {
+      return { action: 'cancel' };
+    }
+
+    if (selected.label.includes('Update Links') || selected.label.includes('Proceed')) {
+      return { action: 'proceed' };
+    } else if (selected.label.includes('Skip Link Updates')) {
+      return { action: 'skip' };
+    } else if (selected.label.includes('Settings')) {
+      return { action: 'settings' };
+    } else if (selected.label.includes('Abort Rename')) {
+      return { action: 'abort' };
```

Comment:
Minor: have you consider adding an `action` field to the quickpick items, and when one is selected to return the value of the field? will spare you all the if/else cases here.